### DISCUSSION TITLE
Multip-Part Form Post

### DISCUSSION THREAD

**Discussion by ServitaGitStu** [nextgenhealthcare/connect#5615] [Q&A]
Hi,

@narupley  I have been trying  build a form post with a file and several other fields,  I have also been replying to the thread below seeking some assistance and just t realise you are probably not monitoring that old forum anymore!!
  
https://forums.mirthproject.io/forum/mirth-connect/support/9583-override-java_home?t=9391

Even using your example workaround the endpoint reports the "file" field is missing.   I am only sending the file currently and expected it to report the other fields are missing.

I have tried this in  3.9.0 and 4.2.0. 

Using your example, I get the error:  "Failed to read the request form. Line length limit 16287 exceeded."

Been reading these references.  (tried both ways)
;https://forums.mirthproject.io/forum/mirth-connect/support/18273-multipart-form-post-of-base64-encoded-pdf-solved#post104394
 ;https://forums.mirthproject.io/forum/mirth-connect/support/9345-mirth-http-multipart-file-name-attribute
 
I even resorted to reading RFCs!!

I found some articles suggesting to include encType but that is when posting from html.  

Thoughts appreciated.  Connector attached. The only real difference is that my source gets a Bas64 string and puts that on the map,  i just decode it back to octets in the connector.

[PEP - Correspondence Example - Custom Multipart HTTP - NARUPLEY.zip](https://github.com/nextgenhealthcare/connect/files/10413566/PEP.-.Correspondence.Example.-.Custom.Multipart.HTTP.-.NARUPLEY.zip)

**Comment by pacmano1** [comment]
@ServitaGitStu , I don't want to speak for Nick but he doesn't chime in often here or on the forums for "support" tickets, he is a full time lead engineer at Nextgen leading the development of Mirth Connect.

**Reply by ServitaGitStu** [reply]
oh really,  he used to,  it was his examples I was using,  or trying to.

**Reply by pacmano1** [reply]
I would suggest you do exactly the version as posted at https://forums.mirthproject.io/forum/mirth-connect/support/18273-multipart-form-post-of-base64-encoded-pdf-solved#post104394

BUT, post this to yourself.

If you have node installed:

`npx http-echo-server`.  This will fire up a dumb http listener on the console (meaning if you are ssh'ed to a server of course) and you can see the exact post as you are sending it.   Compare your output to the one that the forum post creates.   I have used the code suggested in that post, nearly identical.

**Reply by pacmano1** [reply]
https://github.com/watson/http-echo-server is the node thingy

**Reply by pacmano1** [reply]
![image](https://user-images.githubusercontent.com/44065187/212377361-4884953c-56c5-475d-838b-94bff60493e9.png)

**Reply by ServitaGitStu** [reply]
@pacmano1 I have something very similar, main difference is more parts.  Good to know the technique works,  not that I doubted the OP in any way.  Thanks for the tip on posting to a node server,  thouigh that's probably another rabbit hole. 

cheers  have a great weekend :)  I'm going to have beer now or quite soon.

**Reply by pacmano1** [reply]
When you see the difference on the raw posts that the echo server shows, I think it will help a lot.  I often use that technique when a vendor posts a working example in, let's say python, so I do the python version to see exactly what it sends so I can replicate that in Mirth.

**Reply by ServitaGitStu** [reply]
installed node and the "thingy"  you are on-point (as always).  I can see exactly what it sends.  Postman initially,  time for beer now though.

I owe you one too. 

Ta

**Reply by ServitaGitStu** [reply]
Thought I'd feedback a little of my own.

I now have this working pretty much as @pacmano1 linked thread above.  I am using a binary part and multiple text parts.  I'd been through so many changes an attempts that I'd created a boundary to use and then accidentally left a newly generated one in the MultioPartEntityBuilder, which gave me part length exceeded error.  

The http-echo-server was good in that,  that gave me the same error as my real endpoint,  which pointed me back to my MPEB.

**Reply by pacmano1** [reply]
When you get a chance, please close out your forum post.

**Reply by ServitaGitStu** [reply]
This post?  and how doo  you mean close out? 

ta

**Reply by pacmano1** [reply]
You asked the same question on the forum.  You found an answer here.   Post the answer there.

**Reply by ServitaGitStu** [reply]
right

**Comment by 1warpengine** [comment]
I'm sorry but I've tried the method explained here, it works to send PDF in multipart/form-data, but how can you also add to the HTTP Sender contente, more key values? Like I have to add somekey="somevalue", similar what you do with CURL.
